lib/pages/openedListPage.dart

```diff
@@ -0,0 +1,301 @@
+                  return new DropdownMenuItem<String>(
+                    value: value,
+                    child: new Text(value),
+                  );
+                }).toList(),
+              ),
+              SizedBox(
+                height: 5,
+              ),
+              RaisedButton(
+                onPressed: () => addItemToList(),
+                child: Text(
+                  getString(context, 'add_item'),
+                  style: TextStyle(
+                      color: Colors.grey[600], fontWeight: FontWeight.bold),
+                ),
+                color: Colors.white,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _addedItem(BuildContext context, String itemName, String itemCount,
+      String itemType) {
+    return Row(
+      children: [
+        new Container(
+          height: 100,
+          width: MediaQuery.of(context).size.width,
+          child: new Center(
+            child: Stack(
+              children: [
+                Positioned(
+                  top: 25,
+                  left: 10,
+                  child: Container(
+                    height: 50,
+                    width: 50,
+                    child: Image(
+                      image: AssetImage("assets/images/bottle_of_water.png"),
+                    ),
+                  ),
+                ),
+                Positioned(
+                  top: 30,
+                  left: 100,
+                  child: Text(
+                    itemName,
+                    style: TextStyle(
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+                ),
+                Positioned(
+                  top: 60,
+                  left: 100,
+                  child: Text(
+                    itemCount,
+                    style: TextStyle(
+                      color: Colors.grey,
+                    ),
+                  ),
+                ),
+                Positioned(
+                  left: 130,
+                  top: 60,
+                  child: Text(
+                    itemType,
+                    style: TextStyle(
+                      color: Colors.grey,
+                    ),
+                  ),
+                ),
+                Positioned(
+                  top: 40,
+                  right: 10,
+                  child: IconButton(
+                    onPressed: () {},
+                    icon: Icon(Icons.edit),
+                  ),
+                )
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void incrementCounter() {
+    _itemCountController.text = (_itemCount + 1).toString();
+    _itemCount = _itemCount + 1;
+  }
+
+  void decrementCounter() {
+    if (_itemCount > 0) {
+      _itemCountController.text = (_itemCount - 1).toString();
+      _itemCount = _itemCount - 1;
+    }
+  }
+
+  void addItemToList() {
+    setState(() {
```

Comment:
i think setState is not necessary ?

Reply:
If the setState is not used, we need to refresh the page to see the added item